jboss-eap-config-elytron/added/launch/elytron.sh

```diff
@@ -0,0 +1,261 @@
+}
+
+create_elytron_https_connector() {
+    declare name="$1" socket_binding="$2" ssl_context="$3" proxy_address_forwarding="$4"
+    echo "<https-listener name=\"${name}\" socket-binding=\"${socket_binding}\" ssl-context=\"${ssl_context}\" proxy-address-forwarding=\"${proxy_address_forwarding:-true}\"/>"
+}
+
+configure_https() {
+
+  if [ "${CONFIGURE_ELYTRON_SSL}" != "true" ]; then
+    echo "Using PicketBox SSL configuration."
+    return 
+  fi
+
+  local ssl="<!-- No SSL configuration discovered -->"
+  local https_connector="<!-- No HTTPS configuration discovered -->"
+  local missing_msg="WARNING! Partial HTTPS configuration, the https connector WILL NOT be configured. Missing:"
+  local key_password=""
+  local elytron_key_store=""
+  local elytron_key_manager=""
+  local elytron_server_ssl_context=""
+  local elytron_https_connector=""
+
+  if [ -n "${HTTPS_PASSWORD}" -a -n "${HTTPS_KEYSTORE}" -a -n "${HTTPS_KEYSTORE_TYPE}" ]; then
+    if [ -n "${HTTPS_KEY_PASSWORD}" ]; then
+      key_password="${HTTPS_KEY_PASSWORD}"
+    else
+      echo "No HTTPS_KEY_PASSWORD was provided; using HTTPS_PASSWORD for Elytron LocalhostKeyManager."
+      key_password="${HTTPS_PASSWORD}"
+    fi
+
+    local elytron_key_store=$(create_elytron_keystore "LocalhostKeyStore" "${HTTPS_KEYSTORE}" "${HTTPS_PASSWORD}" "${HTTPS_KEYSTORE_TYPE}" "${HTTPS_KEYSTORE_DIR}")
+    local elytron_key_manager=$(create_elytron_keymanager "LocalhostKeyManager" "LocalhostKeyStore" "${key_password}")
+    local elytron_server_ssl_context=$(create_elytron_ssl_context "LocalhostSslContext" "LocalhostKeyManager")
+    local elytron_https_connector=$(create_elytron_https_connector "https" "https" "LocalhostSslContext" "true")
+  else
+    if [ -z "${HTTPS_PASSWORD}" ]; then
+      missing_msg="$missing_msg HTTPS_PASSWORD"
+    fi
+    if [ -z "${HTTPS_KEYSTORE}" ]; then
+      missing_msg="$missing_msg HTTPS_KEYSTORE"
+    fi
+    if [ -z "${HTTPS_KEYSTORE_TYPE}" ]; then
+      missing_msg="$missing_msg HTTPS_KEYSTORE_TYPE"
+    fi
+    echo ${missing_msg}
+  fi
+
+  # check for new config tag, use that if it's present
+  if [ "$(has_elytron_tls "${CONFIG_FILE}")" = "true" ]; then
+    # has new tag, replace it
+    insert_elytron_tls
+  fi
+
+  if [ "$(has_elytron_keystore "${CONFIG_FILE}")" = "true" ]; then
+    sed -i "s|<!-- ##ELYTRON_KEY_STORE## -->|${elytron_key_store}<!-- ##ELYTRON_KEY_STORE## -->|" $CONFIG_FILE
+    sed -i "s|<!-- ##ELYTRON_KEY_MANAGER## -->|${elytron_key_manager}<!-- ##ELYTRON_KEY_MANAGER## -->|" $CONFIG_FILE
+    sed -i "s|<!-- ##ELYTRON_SERVER_SSL_CONTEXT## -->|${elytron_server_ssl_context}<!-- ##ELYTRON_SERVER_SSL_CONTEXT## -->|" $CONFIG_FILE
+  else # legacy config
+    legacy_elytron_tls=$(elytron_legacy_config "${HTTPS_KEYSTORE}" "${HTTPS_KEYSTORE_TYPE}" "${HTTPS_PASSWORD}" "${HTTPS_KEY_PASSWORD}" "${HTTPS_KEYSTORE_DIR}")
+    sed -i "s|<!-- ##TLS## -->|${legacy_elytron_tls}|" $CONFIG_FILE
```

Comment:
Similar to my last comment, in the non-L190 if case, this is replacing a comment with an unusual xml structure.

I think L222/3/4/7 can all stay here but outside any if/else, legacy_elytron_tls gets set to "" at the top, and the L190 block calculates it if appropriate, i.e if [ "$(has_elytron_keystore "${CONFIG_FILE}")" = "false" ].

So L190 block calculates all the stuff and does the insert_elytron_tls if needed, the L202 block warns, and then the rest just seds in the calculated values or "".